src/ir_passes/lower/utils.rs

```diff
@@ -71,14 +85,27 @@ pub(super) fn port_name(
             .src_info
             .as_ref()
             .map(|src| src.ports.get(&idx).unwrap().to_string())
+            .or_else(|| debug.then(|| info_name(comp.get(idx).info, comp)))
             .unwrap_or_else(|| format!("p{}", idx.get())),
         ir::PortOwner::Inv { base, .. } => {
-            base.apply(|p, c| port_name(p, ctx, c), ctx)
+            base.apply(|p, c| port_name(p, ctx, c, debug), ctx)
         }
-        ir::PortOwner::Local => format!("p{}", idx.get()),
+        ir::PortOwner::Local => debug
+            .then(|| info_name(comp.get(idx).info, comp))
+            .unwrap_or_else(|| format!("p{}", idx.get())),
     }
 }
 
+pub(super) fn instance_name<C: Ctx<Instance> + Ctx<Info>>(
```

Comment:
Could we define these methods on some relevant struct so that we don't have to thread through the `debug` argument?